src/client/datascience/kernel-launcher/kernelProcess.ts

```diff
@@ -143,8 +143,9 @@ export class KernelProcess implements IKernelProcess {
         } catch (error) {
             // Make sure to dispose if we never get a heartbeat
             this.dispose().ignoreErrors();
-            traceError('Timed out waiting to get a heartbeat from kernel process.');
-            throw new Error('Timed out waiting to get a heartbeat from kernel process.');
+            const message = 'Timed out waiting to get a heartbeat from kernel process.\n';
+            traceError(message + (error as Error).toString());
+            throw new Error(message + (error as Error).toString());
```

Comment:
There are more errors on this file that just show a hardcoded message. Should we show the error to the user everywhere? The original bug only refers to this error which is the most common, but it should be helpful to show it everywhere.

Also, shouldn't all of those messages be localized?

Reply:
Was this tried with a broken kernel? I think that the issue here is that the error might not have the full info from the output or the usable output might be way too much to show. My Julia kernel was broken recently and the output in the log looks like this: 
![image](https://user-images.githubusercontent.com/812783/100946264-d872e500-34b7-11eb-81d1-f89b60dade8d.png)

I though that we were instead considering pointing users at the output window. Like this error message did:

![image](https://user-images.githubusercontent.com/812783/100946365-048e6600-34b8-11eb-99e3-fb23fcff9ec8.png)

Reply:
If wrapping an error, please use `WrappedError`, as that preserves the stack trace instead of `new Error(...error.toString())`

Also, I agree with @IanMatthewHuff 
We should check if the kernel has died. In `launchAsObservable` as have the code `exeObs.proc.on('exit', (exitCode) => {` & `exeObs.proc.stderr.on('data', (data: Buffer | string) => {`
We should use that information to determine if the kernel has died (same output is available a few lines earlier).

**My suggestion is:**
* If the kerrnel dies while waiting for heart beat, there's no point waiting for 30s (e.g. if the kernel dies in the first 1s). 
This is very easy to replicate. Just create a bad kernel.
(I came across this recently when I broken starting of raw kernels, the only message i got was a timeout error, when the kernel died immediately).
In such a case we should stop waiting for the heart beat and let the user know the kernel died & display a messages such as the one @IanMatthewHuff suggested `View Jupyter log for further details`
* If the kernel heart beat times out, again we can display the stderror message to the user (even if kernel has not exited).